examples/fabric-admin/rpc/RpcClient.cpp

```diff
@@ -101,7 +124,7 @@ CHIP_ERROR AddSynchronizedDevice(chip::NodeId nodeId)
         return CHIP_ERROR_INTERNAL;
     }
 
-    return CHIP_NO_ERROR;
+    return WaitForResponse(addSynchronizedDeviceCall);
```

Comment:
busy-looping stops the CHIP main progress loop - this seems off here.
Why don't we use teh on_response/on_rpc_error to advance a state machine instead?

https://pigweed.dev/pw_rpc/nanopb/#id1

Reply:
We already did, OnAddDeviceResponseCompleted is the on_response and on_rpc_error is nullptr. Yes, it stop the main loop here to prevent from maintaining a more complicated state machine